### DISCUSSION TITLE
ESM-tools not working on JUWELS if Stages/2023

### DISCUSSION THREAD

**Discussion by fernandadialzira** [esm-tools/esm_tools#956] [ESM-Tools support] [answered]
Recently, Juwels has updated most modules according to Stages/2023. ESM-tools work when one loads Stages/2022, but it would be great to be able to use newer versions of other Juwels modules. Will it be some sort of update to ESM-tools to work with Stages/2023?

**Comment by JanStreffing** [comment, marked as answer]
Is anything broken with the 2022 stages? If not I would not switch to 2023. Chances are much higher something is wrong with the new stage than with the one we used for a year. In general I prefer the way this is handled at DKRZ, where modules once installed stay there until the machine is taken down. I used the intel 2017 modules until the end of mistral in 2022.

If you do want to tackle this: 
First step of the upgrade would be to install the eccodes tool chain with the new compiler. I can send you the folder with the tool chain packages.

Second step a new section under:
https://github.com/esm-tools/esm_tools/blob/279c277dcad608fdbcaab0fb82af0363f0886f31/configs/machines/juwels.yaml#L96

Then comes the testing of compiler and runtime.

**Reply by fernandadialzira** [reply]
There is nothing broken with the 2022 Stages, actually. I just saw the message that it was deprecated and thought it could become a problem. But if there is no concern on it, I will just keep loading Stages 2022 :)
